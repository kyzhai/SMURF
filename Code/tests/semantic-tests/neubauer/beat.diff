66a67,68
> 	ID: head h : t :: [a] -> a
> 	ID: tail h : t :: [a] -> [a]
