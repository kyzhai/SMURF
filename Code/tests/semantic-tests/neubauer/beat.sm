a = 4 $+ 4
b = 4 $+ 4.
c = 4. $+ 4.

d = 4 $- 2
e = 4. $- 2
f = 4. $- 2.

g = 4 $* 2
h = 4. $* 2
i = 4. $* 2.

j = 4 $/ 2
k = 4. $/ 2
l = 4. $/ 2.

main = []
