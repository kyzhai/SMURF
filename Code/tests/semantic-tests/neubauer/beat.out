Program: 
	SVardef: 
		ID: a :: Num

	4 $+ 4
	SVardef: 
		ID: b :: Num

	4 $+ 4.
	SVardef: 
		ID: c :: Num

	4. $+ 4.
	SVardef: 
		ID: d :: Num

	4 $- 2
	SVardef: 
		ID: e :: Num

	4. $- 2
	SVardef: 
		ID: f :: Num

	4. $- 2.
	SVardef: 
		ID: g :: Num

	4 $* 2
	SVardef: 
		ID: h :: Num

	4. $* 2
	SVardef: 
		ID: i :: Num

	4. $* 2.
	SVardef: 
		ID: j :: Num

	4 $/ 2
	SVardef: 
		ID: k :: Num

	4. $/ 2
	SVardef: 
		ID: l :: Num

	4. $/ 2.
Global Scope: 
	ID: a :: Num
	ID: b :: Num
	ID: c :: Num
	ID: d :: Num
	ID: e :: Num
	ID: f :: Num
	ID: g :: Num
	ID: h :: Num
	ID: i :: Num
	ID: j :: Num
	ID: k :: Num
	ID: l :: Num
	ID: print :: Unknown
	ID: random :: Unknown

